third_party/istio-1.1.7/values-extras.yaml

```diff
@@ -0,0 +1,127 @@
+# Common settings.
+global:
+  omitSidecarInjectorConfigMap: true
+  istioNamespace: istio-system
+
+  proxy:
+    envoyStatsd:
+      # If enabled is set to true, host and port must also be provided. Istio no longer provides a statsd collector.
+      enabled: false
+      host: # example: statsd-svc.istio-system
+      port: # example: 9125
+
+
+gateways:
+  enabled: true
+
+  istio-ingressgateway:
+    enabled: false
+  istio-egressgateway:
+    enabled: false
+  istio-ilbgateway:
+    enabled: false
+
+  cluster-local-gateway:
+    enabled: true
+    labels:
+      app: cluster-local-gateway
+      istio: cluster-local-gateway
+    replicaCount: 1
+    autoscaleMin: 1
+    autoscaleMax: 1
+    resources: {}
+      # limits:
+      #  cpu: 100m
+      #  memory: 128Mi
+      #requests:
+      #  cpu: 1800m
+      #  memory: 256Mi
+    cpu:
+      targetAverageUtilization: 60
+    loadBalancerIP: ""
+    loadBalancerSourceRanges: {}
+    externalIPs: []
+    serviceAnnotations: {}
+    podAnnotations: {}
+    type: ClusterIP
+    #externalTrafficPolicy: Local #change to Local to preserve source IP or Cluster for default behaviour or leave commented out
+    ports:
+      ## You can add custom gateway ports
+    - port: 80
+      targetPort: 80
+      name: http2
+      # nodePort: 31380
+    - port: 443
+      name: https
+      # nodePort: 31390
+    - port: 31400
+      name: tcp
+      # nodePort: 31400
+    # Pilot and Citadel MTLS ports are enabled in gateway - but will only redirect
+    # to pilot/citadel if global.meshExpansion settings are enabled.
+    - port: 15011
+      targetPort: 15011
+      name: tcp-pilot-grpc-tls
+    - port: 8060
+      targetPort: 8060
+      name: tcp-citadel-grpc-tls
+    # Addon ports for kiali are enabled in gateway - but will only redirect if
+    # the gateway configuration for the various components are enabled.
+    - port: 15029
+      targetPort: 15029
+      name: http2-kiali
+    # Telemetry-related ports are enabled in gateway - but will only redirect if
+    # the gateway configuration for the various components are enabled.
+    - port: 15030
+      targetPort: 15030
+      name: http2-prometheus
+    - port: 15031
+      targetPort: 15031
+      name: http2-grafana
+    - port: 15032
+      targetPort: 15032
+      name: http2-tracing
```

Comment:
Why are all of these ports needed? Seems like we don't set them in the other `values` files.

Reply:
I copied install/kubernetes/helm/istio/example-values/values-istio-gateways.yaml to values-extras.yaml. I will remove the extra values.

My thoughts is to create template or extra files in thirdparty to deploy cluster-local-gateway when I filed the issue. 
It's okay to me that we have a document to describe how to do this rather than generate file here.

Reply:
yu2003w one thing to do is to generate these files using the same `istio-lean.yaml` values, and turn off components (like what install/kubernetes/helm/istio/example-values/values-istio-gateways.yaml) do.

That way we will have it consistent istio-lean.yaml.  (Unfortunately I couldn't find a way to share yaml between the istio-lean.yaml and istio.yaml, there may still be discrepancies between though two files).

Adding another document to describe how to do this is also good.

Reply:
@tcnghia updated ```values-extra.yaml``` to keep consistent with ```Istio-lean.yaml```.
Do you think ```PodDisruptionBudget``` is needed?
```
# Source: istio/charts/gateways/templates/poddisruptionbudget.yaml

apiVersion: policy/v1beta1
kind: PodDisruptionBudget
metadata:
  name: cluster-local-gateway
  namespace: istio-system
  labels:
    chart: gateways
    heritage: Tiller
    release: release-name
    app: cluster-local-gateway
    istio: cluster-local-gateway
spec:

  minAvailable: 1
  selector:
    matchLabels:
      release: release-name
      app: cluster-local-gateway
      istio: cluster-local-gateway
----
```

Reply:
Let's leave out the disruption budgets since I believe we have one pod for these deployment.  Disruption budgets applying to single pod deployment makes it hard to upgrade nodes etc.